samples/drivers/counter/alarm/src/main.c

```diff
@@ -72,6 +72,8 @@ struct counter_alarm_cfg alarm_cfg;
 #define TIMER DT_INST(0, renesas_rz_gtm_counter)
 #elif defined(CONFIG_COUNTER_CC23X0_RTC)
 #define TIMER DT_NODELABEL(rtc0)
+#elif DT_HAS_ALIAS(counter0)
```

Comment:
I believe this should be done only for imx95.
This might impact other boards/platforms that have `counter0` in dts but they don't what to use it for timer.

Reply:
Linue 58 is only valid for CONFIG_COUNTER_AMBIQ) 
```
#elif defined(CONFIG_COUNTER_AMBIQ)
#ifdef TIMER
#undef TIMER
#endif
#define TIMER DT_NODELABEL(counter0)
#elif defined(CONFIG_COUNTER_SNPS_DW)
```

Reply:
It still has some difference, one is node label, the other one is node alias, I find currently no API like DT_HAS_LABEL, maybe need to implement this one API, and the fall-through will be check whether has counter0 label, if yes set TIMER to it, otherwise check whether has counter0 alias, if yes, set COUNTER to it, otherwise, report error.